src/v/kafka/server/handlers/alter_configs.cc

```diff
@@ -123,6 +125,11 @@ create_topic_properties_update(alter_configs_resource& resource) {
                   update.properties.retention_duration, cfg.value);
                 continue;
             }
+            if (cfg.name == topic_property_data_policy) {
+                parse_and_set_optional(
+                  update.properties.data_policy, cfg.value);
+                continue;
```

Comment:
what is the behavior when the json is invalid and parsing fails?

Reply:
In rpk:
Error: EOF
In log:
INFO  2021-09-03 21:53:28,777 [shard 1] kafka - connection_context.cc:308 - Detected error processing request: v8_engine::data_policy_exeption (data_policy.h:74 - json is incorrect)

Reply:
the experience here at the client side should probably do something nicer than print EOF. For example:

```
nwatkins@gordon:~/src/redpanda$ vbuild/go/linux/bin/rpk topic set-config foo retention.ms=some_string
Error: unable to parse property retention.ms value {some_string}
```